### DISCUSSION TITLE
Debug Test for OpenAI

### DISCUSSION THREAD

**Discussion by skperdue68** [mealie-recipes/mealie#4478] [Q&A] [answered]
Ive got Ollama up and running okay and it loos like once (and only once_ I managed to get the debug test for OpenAI to respond back with an analysis of the image, anytime else I'm getting an error 504 which looks like its a timeout.    I'm not suprised this is running slow as I'm testing this all on a pretty slow machine.    I left the timeout value in my environment variables blank at first but then set a rather high value and I'm still getting returned error 504 long before that timeout should hit.   Am I missing something?   Because this is a feature I would love to make use of,

**Comment by michael-genson** [comment]
As long as you have `OPENAI_REQUEST_TIMEOUT` set, that should be sufficient on the Mealie side.

For Ollama, have you tested it on larger inputs outside of Mealie? It sounds like an Ollama issue.  Can you check the Ollama logs and see if it offers any more information besides just the status code 504?

**Comment by skperdue68** [comment]
With your reassurance I dug back into things, it looks like it might have been a networking issues between containers.   I did at least get the admin portion to successfully do a debug, I'll test that a bit more then try the actual create from image, if I have any other issues I'll add adddiional notes on tis, thanks so much!

**Comment by skperdue68** [comment]
So I am able to get the debug to work reliably (I've tried multiple images and image formats) all seem to work, although I have noticed at first it was returning all the recipe data after a few runs it started giving less and less data

When I run the add recipe from image it doesn't seem to complete the request no matter how long I let it run

Here is an example of ollama logs for import from image.   Is it just my machine not being able to handle the request in which case you would thin ollama would come back with some kind of a message about that...

2024-10-31 19:07:47 [GIN] 2024/11/01 - 00:07:47 | 200 | 51.623497792s |      172.18.0.4 | POST     "/v1/chat/completions"

**Reply by michael-genson** [reply, marked as answer]
You need pretty beefy hardware to get image processing to work. In my local tests I couldn't get any usable data out of images using Ollama. Check to see if you can get anything out of an image outside of Mealie, first.

**Reply by skperdue68** [reply]
I'm still fairly new to all this so trying to figure out what to prompt for but tests outside of mealie seem to give me info.     I've been trying to figure out the mealie prompt.     I found one on source code that looks like it but it references a passed json format that don't see

**Comment by greirson** [comment]
@skperdue68 which Ollama model are you using in your config?

**Reply by skperdue68** [reply]
I've tried with both llava and llava-llama.
